src/Schema/migrations/getSyncChanges/index.js

```diff
@@ -45,13 +50,33 @@ export default function getSyncChanges(
   )
   const createdTables = createTableSteps.map((step) => step.schema.name)
 
+  type ChangeColumnsMigrationStep = AddColumnsMigrationStep | DestroyColumnMigrationStep
+
   // $FlowFixMe
-  const addColumnSteps: AddColumnsMigrationStep[] = steps.filter(
-    (step) => step.type === 'add_columns',
+  const changeColumnSteps: ChangeColumnsMigrationStep[] = steps.filter(
+    (step) => step.type === 'add_columns' || step.type === 'destroy_column',
   )
-  const allAddedColumns = addColumnSteps
+
+  // $FlowFixMe
+  const allAddedColumns = changeColumnSteps
     .filter((step) => !createdTables.includes(step.table))
-    .map(({ table, columns }) => columns.map(({ name }) => ({ table, name })))
+    .reduce((addedColumns, step) => {
+      if (!createdTables.includes(step.table)) {
+        switch (step.type) {
+          case 'add_columns':
+            return addedColumns.concat(
+              step.columns.map(({ name }) => ({ table: step.table, name })),
+            )
+          case 'destroy_column':
+            return addedColumns.filter(
+              ({ table, name }) => step.table !== table || step.column !== name,
```

Comment:
this doesn't seem correct to me, and there are no tests for this. can you explain?

Reply:
I had wrong understanding of getSyncChanges. I still have.
- If there is an addColumns for 'table' and 'column' in an earlier step, a destroyColumn in a later step should cancel it out. That's why I unnested them and kept them as a single array.